test/e2e/ginkgo/multiple_cluster_deployment_test.go

```diff
@@ -0,0 +1,294 @@
+				[]metav1.LabelSelector{
+					{MatchLabels: map[string]string{"location-group": "edge"}},
+				},
+				[]ksapi.DownsyncObjectTest{
+					{ObjectSelectors: []metav1.LabelSelector{
+						{MatchLabels: map[string]string{"app.kubernetes.io/name": "nginx"}},
+					}}})
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+
+			ginkgo.By("delete the second bindingpolicy")
+			err := ksWds.ControlV1alpha1().BindingPolicies().Delete(ctx, "nginx-2", metav1.DeleteOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+		})
+
+		ginkgo.It("deletes WEC objects when wds deployment is deleted", func() {
+			util.DeleteDeployment(ctx, wds, ns, "nginx")
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
+
+		ginkgo.It("deletes WEC objects when BindingPolicy is deleted", func() {
+			err := ksWds.ControlV1alpha1().BindingPolicies().Delete(ctx, "nginx", metav1.DeleteOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
+
+		// 1. Create object 1 with label A and label B, object 2 with label B,
+		// and a Placement that matches labels A AND B, and verify that only
+		// object 1 matches and is delivered.
+		// 2. Patch the cluster selector and expect objects to be removed and downsynced correctly.
+		ginkgo.It("downsync objects that fully match on object and cluster selector", func() {
+			ginkgo.By("create two deployments and a bindingpolicy that matches only one")
+			util.DeleteDeployment(ctx, wds, ns, "nginx")
+			util.CreateDeployment(ctx, wds, ns, "one",
+				map[string]string{
+					"label1": "test1",
+				})
+			util.CreateDeployment(ctx, wds, ns, "two",
+				map[string]string{
+					"label1": "test1",
+					"label2": "test2",
+				})
+			util.CreateBindingPolicy(ctx, ksWds, "both-labels",
+				[]metav1.LabelSelector{
+					{MatchLabels: map[string]string{"name": "cluster1"}},
+				},
+				[]ksapi.DownsyncObjectTest{
+					{ObjectSelectors: []metav1.LabelSelector{
+						{MatchLabels: map[string]string{
+							"label1": "test1",
+							"label2": "test2",
+						}},
+					}}})
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+
+			ginkgo.By("patch the cluster selector")
+			patch := []byte(`{"spec": {"clusterSelectors": [{"matchLabels": {"name": "cluster2"}}]}}`)
+			_, err := ksWds.ControlV1alpha1().BindingPolicies().Patch(
+				ctx, "both-labels", types.MergePatchType, patch, metav1.PatchOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+		})
+
+		ginkgo.It("handles OR of cluster and object selectors", func() {
```

Comment:
This is not an accurate description of what is being tested for the cluster label selectors.

Reply:
changed to "handles clusterSelector with MatchLabels and MatchExpressions"